modules/nextflow/src/main/groovy/nextflow/executor/FluxExecutor.groovy

```diff
@@ -0,0 +1,174 @@
+ *
+ *
+ * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ * @author Vanessa Sochat <sochat1@llnl.gov>
+ */
+@Slf4j
+class FluxExecutor extends AbstractGridExecutor {
+
+    static final private Pattern SUBMIT_REGEX = ~/(ƒ.+)/
+
+    /**
+     * Gets the directives to submit the specified task to the cluster for execution
+     *
+     * @param task A {@link TaskRun} to be submitted
+     * @param result The {@link List} instance to which add the job directives
+     * @return A {@link List} containing all directive tokens and values.
+     */
+    protected List<String> getDirectives(TaskRun task, List<String> result) {
+        return result
+    }
+
+    // Flux does not require a special token or header
+    String getHeaderToken() { null }
+
+    /**
+     * The command line to submit this job
+     *
+     * @param task The {@link TaskRun} instance to submit for execution to the cluster
+     * @param scriptFile The file containing the job launcher script
+     * @return A list representing the submit command line
+     */
+    @Override
+    List<String> getSubmitCommandLine(TaskRun task, Path scriptFile ) {
+
+        List<String> result = ['flux', 'mini', 'submit']
+        result << '--setattr=cwd=' + quote(task.workDir)
+        result << '--job-name="' + getJobNameFor(task) + '"'
+        result << '--output=' + quote(task.workDir.resolve(TaskRun.CMD_LOG))  // -o OUTFILE
+
+        if( task.config.cpus > 1 ) {
+            result << '--cores=' + task.config.cpus.toString()
+        }
+
+        // Time limit in minutes when no units provided
+        if( task.config.time ) {
+            result << '--time-limit=' + task.config.getTime().format('mm')
+        }
+
+        // Flux does not support asking for specific memory
+        if( task.config.getMemory() ) {
+            log.debug "Custom memory request is not currently supported by Flux."
+        }
+
+        // the requested partition (a.k.a queue) name
+        if( task.config.queue ) {
+            result << '--queue=' + (task.config.queue.toString())
+        }
+
+        // Any extra cluster options the user wants!
+        // Options tokenized with ; akin to OarExecutor
+        if( task.config.clusterOptions ) {
+
+            // Split by space
+            for (String item : task.config.clusterOptions.toString().tokenize(' ')) {
+                if ( item ) {
+                    result << item.stripIndent().trim()
+                }
+            }
+        }
+        result << '/bin/bash' << scriptFile.getName()
+        return result
+    }
+
+    /**
+     * Parse the string returned by the {@code flux submit} command and extract the job ID string
+     *
+     * @param text The string returned when submitting the job
+     * @return The actual job ID string
+     */
+    @Override
+    def parseJobId(String text) {
+
+        // Parse the special "f" first
+        for( String line : text.readLines() ) {
+            def m = SUBMIT_REGEX.matcher(line)
```

Comment:
The jobid will always be returned in the [F58](https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_19.html) encoding, but depending on locale it may start with an `f` or an `ƒ`. It may be simpler here to just grab the first word printed by `flux mini submit` and return it without trying to match or parse the jobid?

If you want to ensure the jobid validity, you could pass the output to `flux job id <jobid>` which will succeed and return an integer for any valid jobid.

Reply:
I sort of do this by way of first checking for the pattern and falling back to the first thing returned. I figured if people have the defaults the first is more targeted / accurate and the second is a catch all.